src/handlers/request/RequestHandlers.ts

```diff
@@ -0,0 +1,32 @@
+import {EnrichedRequestInit, RequestHandler} from './RequestHandlerInterfaces.js';
+
+/**
+ * @description Method to handle requests
+ * @experimental
+ *
+ * @param {string} url The url of the request
+ * @param {RequestInit} options The options for the initializing of the request
+ * @returns {EnrichedRequestInit} The final request that has been handled
+ */
+export const handleRequest = (
+    url: string,
+    requestInit: RequestInit,
+    requestHandlers?: RequestHandler[],
+): EnrichedRequestInit => {
+    if (!requestHandlers || !requestHandlers?.length) {
+        return {...requestInit, url};
+    }
+
+    const handler = requestHandlers.find(
+        (candidate) =>
+            candidate.hasOwnProperty('canProcess') &&
+            candidate.hasOwnProperty('process') &&
+            candidate.canProcess({...requestInit, url}),
+    );
```

Comment:
I liked that the previous solution allowed us to chain handlers We can have one to log requests, one to add header, one for projects, etc. With this solution we need to do everything in the same handler.

I'm wondering if we should do something more akin to [express application middleware](https://expressjs.com/en/guide/using-middleware.html#middleware.application). They manually call next which makes it easier I think

Reply:
Regarding the first point, it would actually still be possible to chain handlers on the client side. When we want to want to configure the `responseHandlers` property of the configuration, the `withFeatures` method accepts an array of `Feature`, which would allow us to go through a series of handlers. That would avoid doing everything within the same handler.

<img width="748" alt="image" src="https://github.com/coveo/platform-client/assets/105073504/0e8bbad4-cc73-4a43-9d69-09bcb274e7c7">

Reply:
As for the second point, we initially did consider that as an option, as described [here](https://docs.google.com/document/d/1vrIXOi_e6tqL9ehPQ5AjWqQ-S_SgOL4kR8JHKbmjaTA/edit#heading=h.cuj4x8nfauqt). However, ultimately, since there was already some logic in place for the handling of responses, implementing a middleware like that solely for requests wouldn't be complete if  we didn't include the responses either. That would've required us to remove what's in place for the responses to be included in this new middleware. 

Plus, since there was already something in place for response handling, we decided to base ourselves on that for request handling.

Needles to say, what's done in this PR is currently experimental. We haven't discarded the option to create a middleware similar to `express`'s. It's just that for the sake of our feature, it wasn't totally necessary to go that direction.

cc @toofff

Reply:
I'm not sure I get how withFeatures would help us splitting the response handlers, it returns a whole new Platform instance but this instance still will only run one handler, no?

Going back to my example, I was thinking of doing something like that:

```ts
const logger: RequestHandler = {
    canProcess: (requestInit: EnrichedRequestInit): boolean => requestInit.method === 'POST',
    process: (requestInit: EnrichedRequestInit): EnrichedRequestInit => {
        console.log(`POST ${requestInit.url}`);
        return requestInit;
}

const addHeader: RequestHandler = {
    canProcess: (requestInit: EnrichedRequestInit): boolean => true,
    process: (requestInit: EnrichedRequestInit): EnrichedRequestInit => ({
        ...requestInit,
        headers: {
            ...requestInit.headers,
            'X-Coveo-Platform-Client': 'some-value',
        }
    });
}
```
With the current implementation I would need to craft a single handler that does both thing even if they aren't related (you can prove me wrong with the withFeatures stuff, but I can't think of how it would work)

Reply:
As discussed, I'll revert the `handleRequest` method to be chain-able again.